### PR TITLE
Fixed project path issue when project folder is moved and load (#557)

### PR DESCRIPTION
**Pull request by zaryabkhan-rs**
> ### Motivate of the pull request
> - [x] To address an existing issue: <[#557](https://github.com/os-fpga/FOEDAG/issues/557#issue-1325805860)>

> ### Describe the technical details
Project path was saved to project file(.ospr) when create a new project. If project folder is moved to another location that leads to invalid path and gui refuse to run the process, because there is incorrect path to the files.
> #### What is currently done? (Provide issue link if applicable)
> - Removed the path from the project file (.ospr).
>- Project Path points to actual path wherever it is moved. 

> #### Testing steps

>-  Create new project.
>- Verify .ospr file created in the project folder. 
>- Verify source files and constraint files are also added in (<project_Name>.srcs) folder inside the project folder. 
>- Press start button to check all tools like synthesis , packing etc runs fine without errors.
>- Move project folder to another location.
>- Load  project file .ospr from the GUI using "Open Project..." option under File Menu on the main window from the moved folder.
>- Verify You have all files under "Source Tab" on the GUI main window.
>- Again Press start button to check all tools like synthesis , packing etc runs fine without errors.

> #### What does this pull request change?
> - It changes the file handling process.

> ### Which part of the code base require a change
> 
> - [x] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/Main/ProjectFile/ProjectFileLoader.cpp [hunk 1]
```diff
@@ -50,9 +50,7 @@ void ProjectFileLoader::registerComponent(ProjectFileComponent *comp) {
 void ProjectFileLoader::Load(const QString &filename) {
   if (filename.isEmpty()) return;
 
-  QString strTemp = QString("%1/%2%3").arg(Project::Instance()->projectPath(),
-                                           Project::Instance()->projectName(),
-                                           PROJECT_FILE_FORMAT);
+  QString strTemp = QString("/%1").arg(PROJECT_FILE_FORMAT);
```
Comment (KochynVolodymyr) [inline src/Main/ProjectFile/ProjectFileLoader.cpp:53]: I think this should be reverted.

File: src/Main/ProjectFile/ProjectFileLoader.cpp [hunk 2]
```diff
@@ -70,16 +68,13 @@ void ProjectFileLoader::Load(const QString &filename) {
...
     if (type == QXmlStreamReader::StartElement) {
-      if (reader.name() == PROJECT_PROJECT &&
-          reader.attributes().hasAttribute(PROJECT_PATH)) {
-        QString strPath = reader.attributes().value(PROJECT_PATH).toString();
-        QString strName = strPath.mid(
-            strPath.lastIndexOf("/") + 1,
-            strPath.lastIndexOf(".") - (strPath.lastIndexOf("/")) - 1);
-
+      if (reader.name() == PROJECT_PROJECT) {
+        QString projPath = filename.left(filename.lastIndexOf("/"));
```
Comment (KochynVolodymyr) [inline src/Main/ProjectFile/ProjectFileLoader.cpp:72]: please use QFileInfo instead.

File: src/Main/ProjectFile/ProjectFileLoader.cpp (hunk 1 above)
Comment (zaryabkhan-rs) [reply src/Main/ProjectFile/ProjectFileLoader.cpp:53]: But it's only storing "/.ospr" in strTemp.

**Review by KochynVolodymyr — APPROVED** [review]

**Review by ravic-rs — CHANGES_REQUESTED** [review]
Please update the testing steps to be specific about the expected outcome.
_Load project and press start button to run the process. It should runs fine._

**Comment by zaryabkhan-rs** [comment]
> Please update the testing steps to be specific about the expected outcome. _Load project and press start button to run the process. It should runs fine._

Testing steps are updated.

**Pull request merged**